### ISSUE TITLE
Indentation regression (?) in going from 0.23 to 0.24...

### ISSUE THREAD

**Issue by joshburkart** [google/yapf#621]
Thank you so much for making `yapf`!! I love it and it has without exaggeration revolutionized my Python workflow.

Here's my issue: formatting in 0.23.0 (with no `.style.yapf` file -- just the defaults):

```python
        conv_vector = tf.concat(
            [
                self._conv_vector_right_half[::-1],
                self._conv_vector_right_half[1:]
            ],
            axis=0)
```

Looks great. But in 0.24.0:

```python
        conv_vector = tf.concat([
            self._conv_vector_right_half[::-1],
            self._conv_vector_right_half[1:]
        ],
                                axis=0)
```

Gross positioning for `axis=0`! :( I've tried making a `.style.yapf` file -- the only setting that made this look ok (IMO) was changing `dedent_closing_brackets` to `True`, which yields

```python
        conv_vector = tf.concat(
            [
                self._conv_vector_right_half[::-1],
                self._conv_vector_right_half[1:]
            ],
            axis=0
        )
```

but then isn't really what I want elsewhere... I tried looking through [changes since 0.23.0](https://github.com/google/yapf/compare/v0.23.0...v0.24.0) but couldn't figure out what the culprit was... @hugo-ricateau @gwelymernans

**Comment by hugo-ricateau** [comment]
Hi @joshburkart 

Don't know why you mentioned me. The changes I did are not yet officially released, they are planned for `0.25.0`, while you observe that the issue has appeared between `0.23.0` and `0.24.0`.
@gwelymernans will probably be more likely to know what happened.

In the meantime, an easy workaround is to set `split_before_first_argument` to `true`. Reformats as you expect (at least this piece of code):
```
        conv_vector = tf.concat(
            [
                self._conv_vector_right_half[::-1],
                self._conv_vector_right_half[1:]
            ],
            axis=0)
```

**Comment by joshburkart** [comment]
Sorry @hugo-ricateau; I misread the commit history.

But yes!! That was it! Thank you and sorry for the noise.
